pegasus/test/test_v2_user_routes.rb

```diff
@@ -140,6 +148,50 @@ def expected_v2_students_id_hash_for(user)
         assert_equal(expected_v2_students_id_hash_for(FakeDashboard::STUDENT),
                      JSON.parse(@pegasus.last_response.body))
       end
+
+      it 'returns 403 when seeking nonexistent student' do
+        with_role FakeDashboard::ADMIN
+        @pegasus.get "/v2/students/#{FakeDashboard::UNUSED_USER_ID}"
+        assert_equal 403, @pegasus.last_response.status
+      end
+
+      it 'returns 403 when seeking deleted student' do
+        with_role FakeDashboard::ADMIN
+        @pegasus.get "/v2/students/#{FakeDashboard::DELETED_STUDENT[:id]}"
+        assert_equal 403, @pegasus.last_response.status
+      end
+    end
+
+    describe 'POST /v2/students/:id/update' do
+      NEW_STUDENT_NAME = 'Sherry Student'
+
+      it 'returns 403 "Forbidden" when signed in as another student' do
+        with_role FakeDashboard::STUDENT
+        @pegasus.post "/v2/students/#{FakeDashboard::UPDATED_STUDENT[:id]}/update",
+          {name: NEW_STUDENT_NAME}.to_json,
+          'CONTENT_TYPE' => 'application/json;charset=utf-8'
+        assert_equal 403, @pegasus.last_response.status
+      end
+
+      it 'returns 403 "Forbidden" when signed in as unconnected teacher' do
+        with_role FakeDashboard::TEACHER
+        @pegasus.post "/v2/students/#{FakeDashboard::UPDATED_STUDENT[:id]}/update",
+          {name: NEW_STUDENT_NAME}.to_json,
+          'CONTENT_TYPE' => 'application/json;charset=utf-8'
+        assert_equal 403, @pegasus.last_response.status
+      end
+
+      it 'updates student info when signed in as the teacher' do
+        with_role FakeDashboard::TEACHER_WITH_UPDATED
+        @pegasus.post "/v2/students/#{FakeDashboard::UPDATED_STUDENT[:id]}/update",
```

Comment:
I'm not at all familiar with how the Pegasus fixtures work, so I can't really offer any specific comments on how they work in particular. However, in general to provide an auto-rollback transaction using the Sequel db adapter, you would do something like `DASHBOARD_DB.transaction(rollback: :always) { block }`. (See [`shared/test/test_helper.rb`](https://github.com/code-dot-org/code-dot-org/blob/staging/shared/test/test_helper.rb#L59-L61) for an example)